types/src/protocol_spec.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::transaction::{TransactionInfo, TransactionInfoTrait};
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::arbitrary::Arbitrary;
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest_derive::Arbitrary;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+pub trait ProtocolSpec:
+    Clone + Debug + Serialize + Sync + Send + for<'de> Deserialize<'de>
```

Comment:
Replace `for<'de> Deserialize<'de>` with `DeserializeOwned`

Reply:
Trying to remove it entirely, maybe one rule we can use in the future is anything that `serde`s parameterize on specific types like T and A, and things on `<PS: ProtocolSpec>` are those components that are more of doing "logics".

Reply:
Actually, thinking again about this line, `ProtocolSpec` probably shouldn't have any of these bounds since it doesn't really make sense in that its just a collection of associated types, the actual type that impls `ProtocolSpec` should be meaningless